bundle/config/resources/apps.go

```diff
@@ -0,0 +1,75 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/databricks/cli/libs/log"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/marshal"
+	"github.com/databricks/databricks-sdk-go/service/apps"
+)
+
+type App struct {
+	// This represents the id which is the name of the app that can be used
+	// as a reference in other resources. This value is returned by terraform.
+	ID string `json:"id,omitempty" bundle:"readonly"`
+
+	// SourceCodePath is a required field used by DABs to point databricks app source code
+	// on local disk and use it to point to this source code in the app deployment
+	SourceCodePath string `json:"source_code_path"`
+
+	// Config is an optional field which allows configuring the app following Databricks app configuration format like in app.yml.
+	// When this field is set, DABs read the configuration set in this field and write
+	// it to app.yml in the root of the source code folder in Databricks workspace.
+	// If there’s app.yml defined already, it will be overridden.
```

Comment:
We shouldn't silently do this. If someone defines this field but we notice they have an `app.yml` file in their source code path, we can emit an error saying that this conflicts. Either you use `app.yml` and not use this `config` field, or you don't use `app.yml` and use this `config` field.